deploy/operator/ztp/add-remote-nodes-playbook.yaml

```diff
@@ -7,18 +7,24 @@
     - baremetalhosts: "{{ lookup('file', lookup('env', 'REMOTE_BAREMETALHOSTS_FILE')) | from_json }}"
     - infraenv_name: "{{ lookup('env', 'ASSISTED_INFRAENV_NAME') }}"
     - spoke_namespace: "{{ lookup('env', 'SPOKE_NAMESPACE') }}"
+    - cluster_deployment_name: "{{ lookup('env', 'ASSISTED_CLUSTER_DEPLOYMENT_NAME') }}"
     - day2: "true"
-
   tasks:
   - name: create directory for generated CRDs
     file:
       name: generated
       state: directory
-
   - name: write the remote baremetalHost crds
     template:
       src: "baremetalHost.j2"
       dest: "generated/remote_host_manifests.yaml"
-  
+  - name: write the infraEnv crd
+    template:
+      src: "infraEnv-latebinding.j2"
+      dest: "generated/day2-latebinding-infraenv.yaml"
+    when: cluster_deployment_name == ""
+  - name: apply infraenvs with oc
+    ansible.builtin.command: "oc apply -f generated/day2-latebinding-infraenv.yaml"
+    when: cluster_deployment_name == ""
```

Comment:
You could use an ansible `block:` that would have a single `when:` statement encompassing both the template entry and the ansible.builtin.command entry, psuedo:

What you have now:
```
if cdn:
  template()
if cdn:
  oc_apply()
```

What you it should be:
```
if cdn:
  template()
  oc_apply()
```